application/classes/Ushahidi/Repository/Form/Stats.php

```diff
@@ -87,6 +88,80 @@ public function getResponses($form_id)
 			->execute($this->db)
 			->get('total');
 	}
+
+	public function countTotalPending($form_id, $total_sent) {
+		$form_id = intval($form_id);
+		//total_contacts = select count(contact_id) from targeted_survey_state where form_id=8 and survey_status IN ('RECEIVED RESPONSE','PENDING RESPONSE')
+		//total_attributes = select count(form_attributes.id) from form_attributes INNER JOIN form_stages ON form_attributes.form_stage_id = form_stages.id WHERE form_stages.form_id=8
+		$total_contacts = DB::query(Database::SELECT,
+			"select count(contact_id) as total from targeted_survey_state where form_id=$form_id
+			 and survey_status NOT IN ('SURVEY FINISHED')")
+		->execute($this->db)->get('total');
+		$total_attributes = DB::query(Database::SELECT,
+			"select count(form_attributes.id) as total from form_attributes 
+			INNER JOIN form_stages ON form_attributes.form_stage_id = form_stages.id WHERE form_stages.form_id=$form_id")
+			->execute($this->db)->get('total');
+		$internal_query = "SELECT count(form_attributes.form_stage_id) as counted from targeted_survey_state " .
+			"INNER JOIN form_stages ON targeted_survey_state.form_id=form_stages.form_id " .
+			"INNER JOIN form_attributes ON form_stages.id = form_attributes.form_stage_id " .
+			"INNER JOIN (SELECT form_attributes.priority, targeted_survey_state.form_attribute_id, targeted_survey_state.contact_id " .
+			"FROM form_attributes " . "INNER JOIN targeted_survey_state ON form_attributes.id =targeted_survey_state.form_attribute_id " .
+			"WHERE targeted_survey_state.form_id=$form_id and targeted_survey_state.survey_status='INACTIVE'";
+		$query_string = "SELECT SUM(results.counted) as total FROM ($internal_query) as internal_query "
+			. "ON internal_query.contact_id = targeted_survey_state.contact_id "
+			. "WHERE form_attributes.priority > internal_query.priority "
+			. "AND survey_status = 'INACTIVE' "
+			. "AND form_stages.form_id=$form_id "
+			. "group by targeted_survey_state.contact_id, form_attributes.form_stage_id) as results;";
+		$total_pending_for_inactive = DB::query(Database::SELECT,
+		$query_string
+		)->execute($this->db)->get('total');
+		//$total_sent_to_inactive = $total_pending_for_inactive > 0? $total_attributes - $total_pending_for_inactive : 0;
+		return ($total_contacts * $total_attributes) - $total_sent - $total_pending_for_inactive;
```

Comment:
Also, yeah, this is a mind-melting set of queries. 
But for re-use and simplicity of reading, maybe we could split some of these queries into their own methods? There  might be benefits to being get these for use in other stats later.

Reply:
yea. Right now this is posted to discuss the query most since the PR needs to be cleaned up